site/zenodo_rdm/legacy/services.py

```diff
@@ -84,13 +95,20 @@ class LegacyRecordService(RDMRecordService):
     def create(self, identity, data, uow=None, expand=False):
         """Create a draft and prereserve the DOI."""
         res = super().create(identity, data, uow=uow, expand=False)
-        res = self.pids.create(
-            identity=identity,
-            id_=res.id,
-            scheme="doi",
-            expand=True,
-            uow=uow,
-        )
+        pids = data.get("pids", {})
+
+        # If pids is provided, it is created automatically.
+        # TODO confirm this is correct
+        # TODO what happens with other providers / schemes? They are allowed in RDM
```

Comment:
Hopefully we never get from the deserializer other types of PIDs since we control it :)